Source/AwsCommonRuntimeKit/crt/Future.swift

```diff
@@ -0,0 +1,60 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import Foundation
+
+public class Future<Value> {
+    public typealias FutureResult = Result<Value, Error>
+   
+    var _value: FutureResult?
+    
+    var waiter = DispatchSemaphore(value: 0)
+    
+    private var _observers = [((FutureResult) -> Void)]()
+    
+    public init(value: FutureResult? = nil) {
+        self._value = value
+    }
+    
+    public func get() throws -> Value? {
+        waiter.wait()
```

Comment:
What happens if I call get() and the future is already completed?

Reply:
good point, will fix